src/common/animatecmn.cpp

```diff
@@ -28,14 +28,118 @@
 #include "wx/brush.h"
 #include "wx/image.h"
 #include "wx/dcmemory.h"
+#include "wx/module.h"
 
-const char wxAnimationCtrlNameStr[] = "animationctrl";
+#include "wx/gifdecod.h"
+#include "wx/anidecod.h"
+
+#include "wx/private/animate.h"
 
-// global object
+// global objects
+const char wxAnimationCtrlNameStr[] = "animationctrl";
 wxAnimation wxNullAnimation;
 
-wxIMPLEMENT_ABSTRACT_CLASS(wxAnimationBase, wxObject);
-wxIMPLEMENT_ABSTRACT_CLASS(wxAnimationCtrlBase, wxControl);
+wxIMPLEMENT_DYNAMIC_CLASS(wxAnimation, wxObject);
+
+#if !defined(wxHAS_NATIVE_ANIMATIONCTRL)
+    // In this case the "native" ctrl is the generic ctrl. See wx/animate.h
+    wxIMPLEMENT_CLASS(wxAnimationCtrl, wxControl);
```

Comment:
```suggestion
    wxIMPLEMENT_CLASS(wxAnimationCtrl, wxGenericAnimationCtrl);
```

This is needed in generic-only builds when the program uses `wxAnimationCtrl` so the `IsCompatibleWith` works correctly.

Reply:
Will fix, thanks!